applications/zpc/components/zwave_command_classes/src/zwave_command_class_node_info_resolver.c

```diff
@@ -329,6 +330,74 @@ static void on_node_information_update(zwave_node_id_t node_id,
                                      sizeof(node_info->specific_device_class));
 }
 
+static void on_secure_node_info_no_response(attribute_store_node_t secure_nif_node)
+{
+  uint8_t nif[ZWAVE_CONTROLLER_MAXIMUM_COMMAND_CLASS_LIST_LENGTH];
+  uint8_t nif_length = 0;
+
+  if (SL_STATUS_OK
+        == attribute_store_get_node_attribute_value(secure_nif_node,
+                                                    REPORTED_ATTRIBUTE,
+                                                    nif,
+                                                    &nif_length)) {
+    if (nif_length != 0) {
+      return;
+    }
+  }
+
+  sl_log_warning(LOG_TAG, "Failed to get Secure NIF Attribute ID %d", secure_nif_node);
+
+  // Are we trying to resolve our own NIF?
+  attribute_store_node_t node_id_node
+    = attribute_store_get_first_parent_with_type(secure_nif_node, ATTRIBUTE_NODE_ID);
+
+  if (get_zpc_node_id_node() == node_id_node) {
```

Comment:
nitpicking: refactor to avoid creating `node_id_node`

Reply:
compiler will optimize that extra variable ? will it ?